src/main/scala/it/unibo/authsim/library/user/model/User.scala

```diff
@@ -0,0 +1,24 @@
+package it.unibo.authsim.library.user.model
+
+/**
+ * Trait that represent a user
+ */
+trait User:
+  val username: String
+  val password: String
+
+/**
+ * Object that represent a user
+ */
+object User:
+  /**
+   * Apply method for  a user
+   * 
+   * @param _username     name of the user
+   * @param _password     password in clear of the user
+   * @return              a new user
+   */
+  def apply(_username:String, _password:String): User=
+    new User:
+      override val username:String=_username
+      override val password:String=_password
```

Comment:
Function parameters should not start with `_`. Also, anonymous extensions should be avoided, you can use a `private case class` as shown below.

```suggestion
  private case class BasicUser(override val username, override val password) extends User
  
  def apply(username:String, password:String): User = BasicUser(username, password)
```

Reply:
Resolved by 1592d1c